library/modules/bigip_provision.py

```diff
@@ -51,21 +54,37 @@
         For example, changing one module to C(dedicated) requires setting all
         others to C(none). Setting the level of a module to C(none) means that
         the module is not activated.
-      - This parameter is not relevant to C(cgnat) and will not be applied to the
-        C(cgnat) module.
+      - Use c(state) absent to set c(level) to none and de-provision module.
+      - This parameter is not relevant to C(cgnat - pre tmos 15.0) or C(mgmt) and will not be
+        applied to the C(cgnat - pre tmos 15.0) or C(mgmt) module.
     type: str
     choices:
       - dedicated
       - nominal
       - minimum
     default: nominal
+  memory:
+    description:
+      - Sets additional memory for Management Module.
+      - Options include small, medium, large, or integer between 0 and maximum available ram.
+      - Use Large for configurations containing more than 2000 objects, or
+        more specifically, for any configuration that exceeds 1000 objects
+        per 2 GB of installed memory. Changing the Management C(mgmt) size
+        after initial provisioning causes a reprovision operation
+    type: str
+    choices:
+      - small
+      - medium
+      - large
+      - integer between 0 and maximum available ram.
```

Comment:
can you explain this :) cause the choices are not part of arg_spec so this needs to go, if you have non standard values like this, just indicate what users should put in in the documentation, also you can add some sanity checks in the ModuleParameters. DOCUMENTATION string always reflects ArgumentSpec class contents.